android/app/src/main/java/app/organicmaps/MwmActivity.java

```diff
@@ -1109,12 +1119,21 @@ protected void onStop()
   protected void onSafeDestroy()
   {
     super.onSafeDestroy();
-    mLocationPermissionRequest.unregister();
-    mLocationPermissionRequest = null;
-    mLocationResolutionRequest.unregister();
-    mLocationResolutionRequest = null;
-    mPostNotificationPermissionRequest.unregister();
-    mPostNotificationPermissionRequest = null;
+    if (mLocationPermissionRequest != null)
```

Comment:
`mLocationPermissionRequest == null` means that onSafeDestroy() was called twice. Do you have any ideas when and why it happens?

Reply:
No. Check my changes in onSafeCreate
If MwmActivity is launched when car display used -> finish MwmActivity and launch MapPlaceholderActivity

Reply:
ok, but it is still no clear why `if (mLocationPermissionRequest != null)` is needed. `mLocationPermissionRequest` and other such objects are created in onSafeCreate() and destroyed in onSafeDestroy(). It sounds pretty logical to me. Is it possible that onSafeDestroy() is called without onSafeCreate() or twice?

Reply:
I'm finishing `onSafeCreate` before the `mLocationPermissionRequest` is created. That's why it will be null

Reply:
Let's just move creation of requests up.

Reply:
I'd rather move the closing activity to the top of `onSafeCreate`. Do we really need to create something that will be destroyed immediately?

Reply:
> I'd rather move the closing activity to the top of `onSafeCreate`. Do we really need to create something that will be destroyed immediately?

As you wish, but let's keep m*Request deletion without condition. They are marked as NonNull anyway, therefore this extra null check will only generate warnings in Android studio.